api/processors/actions/student/left_project_prematurely.py

```diff
@@ -18,9 +18,9 @@ def process(self) -> None:
         self._set_statuses()
 
     def _update_groups(self) -> None:
-        for group in self.student.groups.all():
-            group.students.remove(self.student)
-            group.students_former.add(self.student)
+        self.student.groups_former.add(*self.student.groups.all())
+        self.student.groups.clear()
+        self.student.save()
```

Comment:
given that it's an m2m relationship, I'm not sure that `.save()` is needed here and in the next changed snippet

Reply:
We have it below also

Reply:
Then I'll approve this PR and create a different issue